package/MDAnalysis/guesser/default_guesser.py

```diff
@@ -0,0 +1,599 @@
+                              ". These can be defined manually using the" +
+                              " keyword 'vdwradii'"))
+
+        lower_bound = kwargs.get('lower_bound', 0.1)
+
+        if box is not None:
+            box = np.asarray(box)
+
+        # to speed up checking, calculate what the largest possible bond
+        # atom that would warrant attention.
+        # then use this to quickly mask distance results later
+        max_vdw = max([vdwradii[t] for t in atomtypes])
+
+        bonds = []
+
+        pairs, dist = distances.self_capped_distance(coords,
+                                                     max_cutoff=2.0 * max_vdw,
+                                                     min_cutoff=lower_bound,
+                                                     box=box)
+        for idx, (i, j) in enumerate(pairs):
+            d = (vdwradii[atomtypes[i]] +
+                 vdwradii[atomtypes[j]]) * fudge_factor
+            if (dist[idx] < d):
+                bonds.append((atoms[i].index, atoms[j].index))
+        return tuple(bonds)
+
+    def add_guessed_bonds(self):
+        vdwradii = self._kwargs.get('vdwradii', None)
+        if not hasattr(self._universe.atoms, 'types'):
+            self._universe.guess_TopologyAttributes(
+                context=self.context, to_guess=['types'])
+
+        return self.guess_bonds(
+            self._universe.atoms,
+            self._universe.atoms.positions,
+            box=self._universe.atoms.dimensions,
+            vdwradii=vdwradii)
+
+    def guess_angles(self, bonds=None):
+        """Given a list of Bonds, find all angles that exist between atoms.
+
+        Works by assuming that if atoms 1 & 2 are bonded, and 2 & 3 are bonded,
+        then (1,2,3) must be an angle.
+
+        Returns
+        -------
+        list of tuples
+            List of tuples defining the angles.
+            Suitable for use in u._topology
+
+
+        See Also
+        --------
+        :meth:`guess_bonds`
+
+      """
+        angles_found = set()
+
+        for b in bonds:
+            for atom in b:
+                other_a = b.partner(atom)  # who's my friend currently in Bond
+                for other_b in atom.bonds:
+                    if other_b != b:  # if not the same bond I start as
+                        third_a = other_b.partner(atom)
+                        desc = tuple(
+                            [other_a.index, atom.index, third_a.index])
+                        # first index always less than last
+                        if desc[0] > desc[-1]:
+                            desc = desc[::-1]
+                        angles_found.add(desc)
+
+        return tuple(angles_found)
+
+    def add_guessed_angles(self):
+
+        if not hasattr(self._universe.atoms, 'bonds'):
+            self._universe.guess_TopologyAttributes(
+                context=self.context, to_guess=['bonds'])
+
+        return self.guess_angles(self._universe.atoms.bonds)
```

Comment:
Here as well, this would add bonds to the universe -- that's maybe less problematic as it's hard to imagine people wanting angles without bonds, and it looks like a lot of the methods rely on them being part of the universe... I think it would still be best to try to avoid affecting the Universe but it would require re-writing the guessing methods a little to only use indices and not atoms.